logic.js

```diff
@@ -0,0 +1,117 @@
+document.addEventListener("DOMContentLoaded", function () {
+  let tasks = [
+    {
+      title: "Complete UI/UX design",
+      desc: "Prepare for the meeting with the client",
+      isDone: false,
+      category: "Freelancer",
+    },
+    {
+      title: "Finish the To-do list logic",
+      desc: "Delete and modify tasks in the list",
+      isDone: true,
+      category: "Training",
+    },
+    {
+      title: "Go shopping",
+      desc: "Buy vegetables, fruits, and milk",
+      isDone: false,
+      category: "Personal",
+    },
+  ];
+
+  const modal = document.getElementById("myModal");
+  const inputField = document.querySelector(".input");
+  const span = document.querySelector(".close");
```

Comment:
use more descriptive naming for these elements, for example modalCloseButton, as a webpage might include mutiple spans